### DISCUSSION TITLE
Making Blocks render better on Smaller Mobile phone screens and event on button press on next block render

### DISCUSSION THREAD

**Discussion by ahmad-punch** [quillforms/quillforms#110] [Q&A] [answered]
I have noticed that the pre-defined blocks do not do well on smaller screens, the content is not centered and moves too far up for a multi-select and the buttons are far down (out of user view port) so when they try to access navigation buttons by swiping down the page changes due to the phone thinking the user swiping down to change page.

How can we edit the UI ? to make sure it works on smaller screens?

Also, I am collecting emails and the forms is quite long, I'd like to collect email by sending request to me server as soon as the user presses next button after entering email and the next block appears. How can I do that? I basically want to add an event listner to the next button that lets the user go to the next block.

**Comment by mdmag** [comment]
@ahmad-punch Please send a Codesandbox to see how your form looks in mobile.
Regarding your second question, please read this [doc](https://github.com/quillforms/quillforms/blob/master/react-docs/available-actions-and-filters.md) to know how to add your own event listeners. In your case, you should use this action: `QuillForms.RendererCore.FieldAnswered`

**Reply by ahmad-punch** [reply]
Thanks, mdmag.

I tried using "`QuillForms.RendererCore.FieldAnswered`" as per the documentation.

Here's my code:

```
import { addAction } from "@wordpress/hooks";
const { Form, useFieldAnswer } = require("@quillforms/renderer-core");


  const emailAnswer = useFieldAnswer("email");


 addAction(
    "QuillForms.RendererCore.FieldAnswered",
    "myOwnAction",
    async function ({ id, label }) {
      console.log("Submiting email to database: ", emailAnswer);

    }
  );
```
  
  And when I press submit, it runs the code like 36 times, not sure why that's happening.
  See screenshot below:
  
![image](https://user-images.githubusercontent.com/62069945/224005935-e9d59100-2e95-467e-90c5-d2cdca33fcda.png)

**Comment by mdmag** [comment]
@ahmad-punch Might be we need to add another action before going to the next question. This will run one time with the last value the user typed. Let me think about it.

**Reply by ahmad-punch** [reply]
Or maybe run only when the user presses the next block button?

**Reply by mdmag** [reply]
@ahmad-punch Going to next question isn't achieved only by pressing on next block button. It can be achieved by mouse wheel swiping or pressing Enter or pressing Tab. It is better to have a general action before going to the next block.

**Comment by mdmag** [comment, marked as answer]
Hello @ahmad-punch,

I gave it another thought and it is really simple.
```js 
import { useCurrentBlock } from "@quillforms/renderer-core";

const App = () => {
    const currentBlock = useCurrentBlock();
    useEffect(() => {

        if(currentBlock?.id === 'MY_BLOCK_ID) {
            // send API Request
        }
    }, [currentBlock] )
}
